### PR TITLE
Remove the SafeBeforeEpoch parameter #2646

### PR DESCRIPTION
**Pull request by jutaro**
Part 1 of issue 2646

### CODE REVIEW

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs [hunk 1]
```diff
@@ -519,7 +506,7 @@ protocolInfoCardano protocolParamsByron@ProtocolParamsByron {
         mkPartialLedgerConfigShelley
           genesisMary
           maxMajorProtVer
-          TriggerHardForkNever
+          TriggerHardForkNever
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs:509]: Stray space

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs [hunk 2]
```diff
@@ -528,10 +515,10 @@ protocolInfoCardano protocolParamsByron@ProtocolParamsByron {
...
     shape :: History.Shape (CardanoEras c)
     shape = History.Shape $ Exactly $
-           K (Byron.byronEraParams     (safeBefore mbLowerBoundShelley) genesisByron)
-        :* K (Shelley.shelleyEraParams (safeBefore mbLowerBoundAllegra) genesisShelley)
-        :* K (Shelley.shelleyEraParams (safeBefore mbLowerBoundMary)    genesisShelley)
-        :* K (Shelley.shelleyEraParams (safeBefore Nothing)             genesisShelley)
+           K (Byron.byronEraParams     (safeBefore Nothing) genesisByron)
+        :* K (Shelley.shelleyEraParams (safeBefore Nothing) genesisShelley)
+        :* K (Shelley.shelleyEraParams (safeBefore Nothing) genesisShelley)
+        :* K (Shelley.shelleyEraParams (safeBefore Nothing) genesisShelley)
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs:521]: Use `History.NoLowerBound` directly and then we can remove `safeBefore`.

**Review by mrBliss — APPROVED** [review]
LGTM

Can you squash the second commit in the first one? It is good practice to address feedback in follow-up commits (like you did), and then before merging we clean up the history of the branch by squashing and rebasing so that we have one or more logical commits.

For example, if you made a typo in one commit and fix it in a second one, there's no added value in including that extra commit in the master branch, just squash it into the first one.

**Comment by jutaro** [comment]
> LGTM
> 
> Can you squash the second commit in the first one? It is good practice to address feedback in follow-up commits (like you did), and then before merging we clean up the history of the branch by squashing and rebasing so that we have one or more logical commits.
> 
> For example, if you made a typo in one commit and fix it in a second one, there's no added value in including that extra commit in the master branch, just squash it into the first one.

I never rewrote a commit history for any repo in my life. So for me you are very special here. Hope nothing gets broken

**Comment by mrBliss** [comment]
> > LGTM
> > Can you squash the second commit in the first one? It is good practice to address feedback in follow-up commits (like you did), and then before merging we clean up the history of the branch by squashing and rebasing so that we have one or more logical commits.
> > For example, if you made a typo in one commit and fix it in a second one, there's no added value in including that extra commit in the master branch, just squash it into the first one.
> 
> I never rewrote a commit history for any repo in my life. So for me you are very special here. Hope nothing gets broken

Squashing commits in a branch is common, rebasing a branch against master, even more so.

File: ouroboros-consensus-cardano-test/test/Test/ThreadNet/Cardano.hs [hunk 3]
```diff
@@ -446,9 +446,7 @@ prop_simple_cardano_convergence TestSetup
                   setupInitialNonce
                   (coreNodes !! fromIntegral nid)
                   ProtocolParamsTransition {
-                      transitionLowerBound =
-                        guard setupByronLowerBound *> Just numByronEpochs
-                    , transitionTrigger = TriggerHardForkAtVersion shelleyMajorVersion
+                    transitionTrigger = TriggerHardForkAtVersion shelleyMajorVersion
```
Comment (mrBliss) [inline ouroboros-consensus-cardano-test/test/Test/ThreadNet/Cardano.hs:449]: Please indent two more spaces :slightly_smiling_face:

**Comment by mrBliss** [comment]
bors merge

**Pull request merged**